File: frontend/src/types/analytics.ts
export interface Overview {
  total_deputados: number;
  total_despesas: number;
  total_proposicoes: number;
  total_votacoes: number;
}

export interface WordCloudItem {
  text: string;
  value: number;
}

export interface Tema {
  tema: string;
  qtd_total: number;
  ano_minimo?: number;
  ano_maximo?: number;
}

export interface Escolaridade {
  escolaridade: string;
  qtd_total: number;
}

export interface EscolaridadeEventos {
  escolaridade: string;
  quantidade_deputados: number;
  media_presenca: number;
}

export interface EscolaridadeFidelidade {
  escolaridade: string;
  quantidade_deputados: number;
  fidelidade_media: number;
}

export interface EscolaridadeProposicoes {
  escolaridade: string;
  total_deputados:  number;
  media_proposicoes: number;
}

export interface EscolaridadeGastos {
  escolaridade: string;
  media_gasto: number;
  quantidade_deputados: number;
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class OverviewResponse(BaseModel):
    total_deputados: int
    total_despesas: int
    total_proposicoes: int
    total_votacoes: int


class TemaResponse(BaseModel):
    tema: str
    qtd_total: int
    ano_minimo: int | None = None
    ano_maximo: int | None = None

class EscolaridadeResponse(BaseModel):
    escolaridade: str | None = None
    qtd_total: int

class EscolaridadeGastosResponse(BaseModel):
    escolaridade: str
    media_gasto: float
    quantidade_deputados: int

class WordCloudResponse(BaseModel):
    text: str
    value: int

class PartidoResponse(BaseModel):
    partido: str
    qtd_total: int

class EscolaridadeFidelidadeResponse(BaseModel):
    escolaridade: str | None = None
    fidelidade_media: float

class EscolaridadeProposicoesResponse(BaseModel):
    escolaridade: str | None = None
    total_deputados: int
    media_proposicoes: float

class EscolaridadeEventosResponse(BaseModel):
    escolaridade: str | None = None
    total_deputados: int
    media_presenca: float